manager/src/main/java/com/redhat/service/smartevents/manager/api/models/requests/ProcessorRequest.java

```diff
@@ -42,13 +44,29 @@ public ProcessorRequest(String name, Action action) {
         this.action = action;
     }
 
+    public ProcessorRequest(String name, Source source) {
+        this.name = name;
+        this.source = source;
+    }
+
     public ProcessorRequest(String name, Set<BaseFilter> filters, String transformationTemplate, Action action) {
         this.name = name;
         this.filters = filters;
         this.transformationTemplate = transformationTemplate;
         this.action = action;
     }
 
+    @JsonIgnore
+    public ProcessorType getType() {
+        if (getSource() != null) {
+            return ProcessorType.SOURCE;
+        }
+        if (getAction() != null) {
+            return ProcessorType.SINK;
+        }
+        return null;
```

Comment:
Wouldn't it have sense to throw  an exception in here (fail fast)?
If the processor type cannot be resolved than IMHO something is wrong and we shouldn't continue.

Reply:
Thanks for the hint. Done :+1: